src/app/beer_garden/scheduler.py

```diff
@@ -670,6 +692,22 @@ def remove_job(job_id: str) -> None:
     # The scheduler takes care of removing the Job from the database
     return db.query_unique(Job, id=job_id)
 
+@publish_event(Events.JOB_EXECUTED)
+def execute_job(job_id: str, reset_interval=False) -> Job:
+    """Execute a Job ad-hoc
+
+    Creates a new job with a trigger for now.
+
+    Args:
+        job_id: The Job ID
+        reset_interval: Whether executing should reset the job interval trigger
+
+    Returns:
+        The spawned Request
+    """
+    logger.info('Job executed from routing')
+
+    return db.query_unique(Job, id=job_id)
```

Comment:
Given the way this works, this may be the spot where you need to use `reset_interval` to set the value of `reschedule_on_finish` on the Job.  Maybe.  If it's more complicated than that then it might not be worth supporting the `reset_interval` thing.